src/solc_0.8.13/common/OperatorFilterer/UpdatableOperatorFiltererUpgradeable.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import {Initializable} from "@openzeppelin/contracts-upgradeable-0.8.13/proxy/utils/Initializable.sol";
+
+import {IOperatorFilterRegistry} from "./IOperatorFilterRegistry.sol";
+
+abstract contract UpdatableOperatorFiltererUpgradeable is Initializable {
+    error OperatorNotAllowed(address operator);
+    error OnlyOwner();
+
+    IOperatorFilterRegistry public operatorFilterRegistry;
+
+    function __UpdatableOperatorFiltererUpgradeable_init(
+        address _registry,
+        address subscriptionOrRegistrantToCopy,
+        bool subscribe
+    ) internal onlyInitializing {
+        operatorFilterRegistry = IOperatorFilterRegistry(_registry);
+        // If an inheriting token contract is deployed to a network without the registry deployed, the modifier
+        // will not revert, but the contract will need to be registered with the registry once it is deployed in
+        // order for the modifier to filter addresses.
+        if (address(operatorFilterRegistry).code.length > 0) {
+            if (subscribe) {
+                operatorFilterRegistry.registerAndSubscribe(address(this), subscriptionOrRegistrantToCopy);
+            } else {
+                if (subscriptionOrRegistrantToCopy != address(0)) {
+                    operatorFilterRegistry.registerAndCopyEntries(address(this), subscriptionOrRegistrantToCopy);
+                } else {
+                    operatorFilterRegistry.register(address(this));
+                }
+            }
+        }
+    }
+
+    modifier onlyAllowedOperator(address from) virtual {
+        // Check registry code length to facilitate testing in environments without a deployed registry.
+        if (address(operatorFilterRegistry).code.length > 0) {
+            // Allow spending tokens from addresses with balance
+            // Note that this still allows listings and marketplaces with escrow to transfer tokens if transferred
+            // from an EOA.
+            if (from == msg.sender) {
+                _;
+                return;
+            }
+            if (!operatorFilterRegistry.isOperatorAllowed(address(this), msg.sender)) {
+                revert OperatorNotAllowed(msg.sender);
+            }
+        }
+        _;
+    }
+
+    modifier onlyAllowedOperatorApproval(address operator) virtual {
+        // Check registry code length to facilitate testing in environments without a deployed registry.
+        if (address(operatorFilterRegistry).code.length > 0) {
+            if (!operatorFilterRegistry.isOperatorAllowed(address(this), operator)) {
+                revert OperatorNotAllowed(operator);
+            }
+        }
```

Comment:
Why we just not using `&&` to do this

Reply:
`UpdatableOperatorFiltererUpgradeable.sol` code has been taken from OpenSea registry and modified as least possible in order to minimize alterations and have OpenSea compliant filter. If/when our pull request to their filter registry is approved, this will be removed.